File: app/src/main/java/com/sakuna63/tumbin/application/fragment/LoginFragment.java
package com.sakuna63.tumbin.application.fragment;

import android.content.Intent;
import android.databinding.DataBindingUtil;
import android.net.Uri;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.customtabs.CustomTabsIntent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.hannesdorfmann.fragmentargs.annotation.FragmentWithArgs;
import com.sakuna63.tumbin.R;
import com.sakuna63.tumbin.application.activity.Henson;
import com.sakuna63.tumbin.application.contract.LoginContract;
import com.sakuna63.tumbin.databinding.FragmentLoginBinding;

@FragmentWithArgs
public class LoginFragment extends BaseFragment implements LoginContract.View {
    public static final String TAG = LoginFragment.class.getSimpleName();

    private LoginContract.Presenter presenter;
    private FragmentLoginBinding binding;

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater,
                             @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        binding = DataBindingUtil.inflate(inflater, R.layout.fragment_login, container, false);
        return binding.getRoot();
    }

    @Override
    public void onViewCreated(View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        binding.setPresenter(presenter);
        presenter.init();
    }

    @Override
    public void setPresenter(LoginContract.Presenter presenter) {
        this.presenter = presenter;
    }

    @Override
    public void setLoginButtonActive(boolean active) {
        binding.buttonLogin.setEnabled(active);
    }

    @Override
    public void setLoginProgress(boolean visible, @NonNull String message) {
        binding.containerProgress.setVisibility(visible ? View.VISIBLE : View.GONE);
        binding.progressbar.setVisibility(View.VISIBLE);
        binding.textLoginProgress.setText(message);
    }

    @Override
    public void showErrorMessage(@NonNull String message) {
        binding.containerProgress.setVisibility(View.VISIBLE);
        binding.progressbar.setVisibility(View.GONE);
        binding.textLoginProgress.setText(message);
    }

    @Override
    public void navigateToHome() {
        getActivity().finish();
        Intent intent = Henson.with(getContext()).gotoHomeActivity().build();
        startActivity(intent);
    }

    @Override
    public void navigateToLoginPage(@NonNull String url) {
        CustomTabsIntent tabsIntent = new CustomTabsIntent.Builder().build();
        tabsIntent.launchUrl(getActivity(), Uri.parse(url));
    }

    public void onNewIntent(Intent intent) {
        presenter.onLoginCallback(intent.getDataString());
    }
}

File: app/src/main/java/com/sakuna63/tumbin/application/fragment/LoginFragment.kt
package com.sakuna63.tumbin.application.fragment

import android.content.Intent
import android.databinding.DataBindingUtil
import android.net.Uri
import android.os.Bundle
import android.support.customtabs.CustomTabsIntent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.sakuna63.tumbin.R
import com.sakuna63.tumbin.application.activity.HomeActivity
import com.sakuna63.tumbin.application.contract.LoginContract
import com.sakuna63.tumbin.databinding.FragmentLoginBinding

class LoginFragment : BaseFragment(), LoginContract.View {

    lateinit private var presenter: LoginContract.Presenter
    lateinit private var binding: FragmentLoginBinding

    override fun onCreateView(inflater: LayoutInflater?,
                              container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        binding = DataBindingUtil.inflate<FragmentLoginBinding>(inflater!!,
                R.layout.fragment_login, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View?, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        binding.presenter = presenter
        presenter.init()
    }

    override fun setPresenter(presenter: LoginContract.Presenter) {
        this.presenter = presenter
    }

    override fun setLoginButtonActive(active: Boolean) {
        binding.buttonLogin.isEnabled = active
    }

    override fun setLoginProgress(visible: Boolean, message: String) {
        binding.containerProgress.visibility = if (visible) View.VISIBLE else View.GONE
        binding.progressbar.visibility = View.VISIBLE
        binding.textLoginProgress.text = message
    }

    override fun showErrorMessage(message: String) {
        binding.containerProgress.visibility = View.VISIBLE
        binding.progressbar.visibility = View.GONE
        binding.textLoginProgress.text = message
    }

    override fun navigateToHome() {
        activity.finish()
        startActivity(Intent(context, HomeActivity::class.java))
    }

    override fun navigateToLoginPage(url: String) {
        val tabsIntent = CustomTabsIntent.Builder().build()
        tabsIntent.launchUrl(activity, Uri.parse(url))
    }

    fun onNewIntent(intent: Intent) {
        presenter.onLoginCallback(intent.dataString)
    }

    companion object {
        val TAG = LoginFragment::class.java.simpleName!!

        fun newInstance() = LoginFragment()
    }
}
